### ISSUE TITLE
NEXT_REDIRECT error banner when using redirect() in Turbopack in dev mode, though the redirect still works fine

### ISSUE THREAD

**Issue by joulev** [vercel/next.js#52227]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #48-Ubuntu SMP Tue Jun 20 20:34:08 UTC 2023
    Binaries:
      Node: 18.16.0
      npm: 9.5.1
      Yarn: 1.22.19
      pnpm: 8.6.6
    Relevant Packages:
      next: 13.4.9-canary.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Routing (next/router, next/navigation, next/link), Turbopack (--turbo)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/joulev/debug/tree/nextjs-turbopack-redirect-error-banner

### To Reproduce

* `next dev` in the linked reproduction repository

Or,

```tsx
// app/page.tsx
import { redirect } from "next/navigation";

export default function Page() {
  redirect("/test");
}
```

```tsx
// app/test/page.tsx
export default function Page() {
  return <div>Hello world</div>;
}
```

and run `/` in dev mode with `--turbo`.

### Describe the Bug

The `redirect` does work, but not before a `NEXT_REDIRECT` error banner shows up.

https://github.com/vercel/next.js/assets/44609036/2c21e0f4-0d75-4fda-8af1-5de355f4b7ae

This is only in dev mode (of course). In prod mode everything works fine.

### Expected Behavior

No `NEXT_REDIRECT` error banners.

**Comment by jhfgloria** [comment]
+1

**Comment by xCoBaLTz** [comment]
+1

**Comment by Kiicchan** [comment]
+1

**Comment by amit548** [comment]
![image](https://github.com/vercel/next.js/assets/36206377/7b5b6749-358b-4347-94f4-9e4cf5d196c7)
![image](https://github.com/vercel/next.js/assets/36206377/136cca0e-e224-46c0-97ba-24a47a6910cc)

**Same Error**

**Comment by imajayt** [comment]
Getting same error without using Turbopack as well.. i am using latest version of NextJS that is 13.4.12 
![image](https://github.com/vercel/next.js/assets/58980115/20aa687e-cd80-420e-af9c-f662961fe8f4)

**Comment by imajayt** [comment]
I am not using Turbopack but still getting error, I am using latest version of NextJS that is 13.4.12:
![image](https://github.com/vercel/next.js/assets/58980115/6790f5bd-19a7-4f12-b884-22522ce4b42b)

**Comment by jeffreys-cat** [comment]
Have the same problem

**Comment by magoz** [comment]
Same error here without Turbopack.

**Comment by magoz** [comment]
This is probably causes because:

> [shuding](https://github.com/shuding) commented [on Jun 8](https://github.com/vercel/next.js/issues/50757#issuecomment-1582239798)
This is by design as documented here: https://nextjs.org/docs/app/api-reference/functions/redirect#redirect. redirect() will throw as a signal to indicate Next.js to stop continuing executing the route below it as it's already redirected.


https://github.com/vercel/next.js/issues/50757#issuecomment-1582239798

**Comment by joulev** [comment]
`redirect` indeed throws an exception, but if non-Turbopack doesn’t show any error banners and Turbopack does, it’s more likely than not that this is a bug in `next dev --turbo`

**Comment by padmaia** [comment]
This appears to be fixed with v13.5+
